dotnet/src/SemanticKernel/SkillDefinition/DotNetNativeFunction.cs

```diff
@@ -0,0 +1,861 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.Globalization;
+using System.Linq;
+using System.Reflection;
+using System.Text.Json;
+using System.Text.RegularExpressions;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Abstractions;
+using Microsoft.SemanticKernel.AI.TextCompletion;
+using Microsoft.SemanticKernel.Diagnostics;
+using Microsoft.SemanticKernel.Orchestration;
+using Microsoft.SemanticKernel.SemanticFunctions;
+
+namespace Microsoft.SemanticKernel.SkillDefinition;
+
+#pragma warning disable format
+
+/// <summary>
+/// Standard Semantic Kernel callable function.
+/// SKFunction is used to extend one C# <see cref="Delegate"/>, <see cref="Func{T, TResult}"/>, <see cref="Action"/>,
+/// with additional methods required by the kernel.
+/// </summary>
+[DebuggerDisplay("{DebuggerDisplay,nq}")]
+internal sealed class DotNetNativeFunction : ISKFunction, IDisposable
```

Comment:
ChatGPT plugins would now be considered a DotNetNativeFunction even if they are pointing to code that is not running DotNet -- might be confusing.

Reply:
The ChatGPT functions are (currently) calling dotnet code - they are dotnet delegate functions.

A future refactor we can plan for may have them further split into their own custom implementation of ISKFunction, but for now they're using this one.

Remember, this is an internal implementation class, not part of the public SDK. I'm not worried about confusing ourselves.

Reply:
I'd vote for just NativeFunction and SemanticFunction since that's what we've always used internally. But not a blocking vote.